File: src/translator.py
from googletrans import Translator

translator = Translator()

async def translate_to_arabic(text):
    """Translates text to Arabic"""
    if not text or len(text.strip()) == 0:
        return text
    try:
        result = translator.translate(text, src_language='auto', dest_language='ar')
        return result['text']
    except Exception as e:
        print(f"Translation error: {e}")
        return text

File: src/src/translator.js
const translate = require('google-translate-api-x').default;

async function translateToArabic(text) {
    try {
        if (!text || text.trim().length === 0) return text;
        
        const result = await translate(text, { to: 'ar' });
        return result.text;
    } catch (error) {
        console.error('Translation error:', error);
        return text;
    }
}

async function translateObject(obj) {
    if (typeof obj === 'string') {
        return await translateToArabic(obj);
    } else if (Array.isArray(obj)) {
        return Promise.all(obj.map(item => translateObject(item)));
    } else if (typeof obj === 'object' && obj !== null) {
        const translated = {};
        for (const [key, value] of Object.entries(obj)) {
            translated[key] = await translateObject(value);
        }
        return translated;
    }
    return obj;
}

module.exports = { translateToArabic, translateObject };
